### PR TITLE
fix: password validation

### LINKED ISSUE #1043: [BUG] No Password validation rules
### Description

in signing up as a user and as a club, in both forms there is no validation for passwords, the weakest passwords like 11111111 , 12345678 get accepted by it.

### Screenshots

_No response_

### Browser 🥦

Google Chrome

### Checklist ✅

- [X] I checked and didn't find similar [issue](https://github.com/MilanCommunity/Milan/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/MilanCommunity/Milan/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (blank for no).

### PR DESCRIPTION
**Pull request by pamechayash**
closes #1043 

Steps that I have taken 
1. I made a new util named ProjectValidations.js, which contains all logic to validate passwords using regex.
2. then use this util in UserRegister and ClubRegister to validate the password before submitting to the backend

my changes will validate the password and check if the main password matches the confirm password.

### CODE REVIEW

**Comment by tamalCodes** [comment]
Please change the branch to `beta` from `main`

**Comment by pamechayash** [comment]
branch changed to beta

File: src/utils/PasswordValidation.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    };
+  } else if (!props.password.match(upperCase)) {
+    return {
+      auth: false,
+      message: "Password should contains uppercase letters!",
+    };
+  } else if (!props.password.match(numbers)) {
+    return { auth: false, message: "Password should contains numbers also!" };
+  } else if (props.password.length < 8) {
+    return { auth: false, message: "Password should not be less than 8" };
```
Comment (tamalCodes) [inline src/utils/PasswordValidation.js:25]:
```suggestion
    return { auth: false, message: "Password should not be less than 8 characters" };
```

File: src/utils/PasswordValidation.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  } else if (!props.password.match(special)) {
+    return {
+      auth: false,
+      message: "Password should contains atleast one special characters",
```
Comment (tamalCodes) [inline src/utils/PasswordValidation.js:29]:
```suggestion
      message: "Password should contain atleast one special character",
```

**Review by tamalCodes — CHANGES_REQUESTED** [review]

**Comment by tamalCodes** [comment]
Also as said previously there are some minor conflicts as some files have been removed. 
Please take a look at #1041 to have a better understanding of the changes that were done/

**Comment by tamalCodes** [comment]
### Let's catch up !
Hi @pamechayash , it has been a while since we last had a conversation so I wanted to know if you are still working on this. Please drop a reply shortly so that we know you are active, feel free to convey any problem you face in the comments, or mail me at tamalcodes@gmail.com. 

> [!WARNING]  
> The PRs are generally closed by 3 weeks if there has been no activity.

**Pull request closed without merging**